core/src/main/java/dk/alexandra/fresco/framework/sce/SecureComputationEngine.java

```diff
@@ -4,55 +4,78 @@
 import dk.alexandra.fresco.framework.builder.ProtocolBuilder;
 import dk.alexandra.fresco.framework.network.Network;
 import dk.alexandra.fresco.framework.sce.resources.ResourcePool;
+import java.time.Duration;
 import java.util.concurrent.Future;
 
 /**
- * Core class of the fresco system, this must be initialized with a concrete ResourcePool
- * from the corresponding protocol suite and the types version of the ProtocolBuilder - hereafter
- * the SecureComputationEngine takes the protocol and runs the matching Application with the
+ * Core class of the fresco system, this must be initialized with a concrete ResourcePool from the
+ * corresponding protocol suite and the types version of the ProtocolBuilder - hereafter the
+ * SecureComputationEngine takes the protocol and runs the matching Application with the
  * corresponding types (i.e. numeric or binary).
  *
  * @param <ResourcePoolT> the resource pool
- * @param <Builder> the typed version of the builder
+ * @param <BuilderT> the typed version of the builder
  */
-public interface SecureComputationEngine<ResourcePoolT extends ResourcePool, Builder extends ProtocolBuilder> {
+public interface SecureComputationEngine<ResourcePoolT extends ResourcePool,
+    BuilderT extends ProtocolBuilder> {
+
+  static final Duration DEFAULT_TIMEOUT = Duration.ofMinutes(10);
 
   /**
    * Executes an application based on the current configuration.
+   * <p>
+   * By default this method times out after waiting 10 minutes for output.
+   * </p>
    *
-   * @param application The application to evaluate.
-   * @param network the network to run the computations on
+   * @param application the application to evaluate
+   * @param resources the resource pool used in evaluation
+   * @param network the network
```

Comment:
Maybe add that the network is expected to be connected before running this method.

Reply:
Hmm. that actually goes for every parameter - they need to be in "working condition"
A bit superfluous I would say

Reply:
Well, that may be correct. Since there is no "connect" method in the network interface anyways, there may not be much point in pointing it out.. Still though, I would add it. I try to expect users to be not that smart :)

Reply:
Yeah, I suppose you would get some sort of network related exception in the case of a non-connected network, which is as can be expected.